src/lib/helpers.ts

```diff
@@ -185,10 +186,21 @@ export const smallFormatDate = (date: Date, locale?: string) => {
 
 export const isSSRMode = typeof window === 'undefined';
 
-export const suggestNewAddress = (addresses?: IWalletAddress[]) => {
+export const suggestNewAddress = (
+	addresses: IWalletAddress[],
+	chain: Chain | NonEVMChain,
+) => {
 	if (!addresses || addresses.length < 1) return '';
 	const isSame = compareAddressesArray(addresses.map(a => a.address));
 	if (isSame) {
+		// Don't suggest EVM addresses for Solana input
+		if ('chainType' in chain && chain.chainType === ChainType.SOLANA) {
```

Comment:
Cant we change it to 
`if (chain.chainType === ChainType.SOLANA)` ?

Reply:
Likewise